libraries/stdlib/stdlib_defs.mtlx

```diff
@@ -1024,6 +1024,17 @@
     <output name="out" type="float" />
   </nodedef>
 
+  <!--
+    Node: <checkerboard>
+    A 2D checkerboard pattern.
+  -->
+  <nodedef name="ND_checkerboard_float" node="checkerboard" nodegroup="procedural2d">
```

Comment:
@crydalch The design of this node seems very reasonable to me, though it seems worthwhile to consider the idea of returning a `color3` value instead of a float, with the two colors of the checkerboard specified as `color1` and `color2` inputs (defaulting to white and black).

I don't have a strong opinion on which design is better for MaterialX, but for reference here are a few examples of `checkerboard` nodes that use the color1/color2 approach:

https://docs.unity3d.com/Packages/com.unity.shadergraph@6.9/manual/Checkerboard-Node.html
https://learn.foundry.com/mari/Content/reference_guide/nodes/Checkerboard.html
https://help.autodesk.com/view/ARNOL/ENU/?guid=arnold_user_guide_ac_texture_shaders_ac_texture_checkerboard_html

Reply:
It would be nice to cover a reasonable "maximum" possible flexibility (which seems to be Arnold) and anything that supports less can restrict what inputs are modifiable / exposed. e.g. frequency in u and v allows for "stripe" textures to be created. A "customized" version of this would sync u and v values. Does this sound reasonable?

Reply:
I like this idea @jstone-lucasfilm

An early version of the node did have color inputs/outputs, but I don't recall why I went away from that... but regardless, I'll make that change. Thank you for the suggestion and references.